crates/chain/tests/test_indexed_tx_graph.rs

```diff
@@ -99,6 +99,97 @@ fn insert_relevant_txs() {
     assert_eq!(graph.initial_changeset(), initial_changeset);
 }
 
+/// Ensure that [`IndexedTxGraph::batch_insert_relevant`] adds transactions that are direct
+/// conflicts with transactions in our graph but are not directly relevant to it.
+///
+/// The graph contains three transactions (A, B, and conflicting_tx), with A and B being relevant
+/// because B spends from A. This test verifies that `conflicting_tx` is inserted into the graph
+/// solely because it is directly conflicting with B.
+#[test]
+fn insert_relevant_conflicting_txs() {
```

Comment:
I think we have better clarity with these tests. I propose having multiple test cases.

### For all these test cases, reuse the following 3 transactions

```
sender_prev_tx ---> send_tx
               \__> undo_send_tx
```

* `sender_prev_tx` has 1 input and 1 output. It is not relevant to the receiver's wallet.
* `send_tx` is the initial tx that was meant to pay the receiver. It has an input spending `sender_prev_tx` and an output which is tracked by the receiver's indexer.
* `undo_send_tx` replaces `send_tx` and sends the tx back to the sender.

### For all these test cases, have more meaningful variable names and data types

* 1 spk for the sender and 1 spk for the receiver. Name them `sender_spk` and `receiver_spk`.
* Use `SpkTxOutIndex` instead of `KeychainTxOutIndex`. Derivation is not important for the tests.

### Test case 1

* Given: An empty receiver `IndexedTxGraph`.
* When: `batch_insert_relevant_unconfirmed` is called with `[send_tx, undo_send_tx]` or `[undo_send_tx, send_tx]`.
* Assert: Both txs are included in the changeset no matter which order we input the txs.

### Test case 2

* Given: A receiver `IndexedTxGraph` which already contains `send_tx`.
* When: `batch_insert_relevant_unconfirmed` is called with `undo_send_tx`.
* Assert: `undo_send_tx` is included in the returned changeset.
* Variations: Try with `batch_insert_relevant`.